lib/MySQL_HostGroups_Manager.cpp

```diff
@@ -4504,6 +4576,213 @@ void MySQL_HostGroups_Manager::read_only_action(char *hostname, int port, int re
 	free(query);
 }
 
+/**
+ * @brief New implementation of the read_only_action method that does not depend on the admin table.
+ *   The method checks each server in the provided list and adjusts the servers according to their corresponding read_only value.
+ *   If any change has occured, checksum is calculated.
+ *
+ * @param mysql_servers List of servers having hostname, port and read only value.
+ * 
+ */
+void MySQL_HostGroups_Manager::read_only_action_v2(const std::list<std::tuple<std::string,int,int>>& mysql_servers) {
```

Comment:
After checking PR #4144, I have seen that we are going to rely on different representation of servers (based on `std::tuple`) for the actions. Since `std::tuple` can't make use of names for fields, it's harder to tell just by type which will be the target usage of each of the tuple fields (other than checking the code or reading the doc). Specially if we start having different methods that receives different tuples encode similar meanings (in this case, servers representation).

```
std::tuple<std::string,int,int> -> here for this PR.
std::tuple<int, std::string, unsigned int, int> -> for replication_lag actions
```

Instead of just adding doc to the function signature, I think that we could use conventions that we have already used in the past for this scenarios like:

- Defining a new type for these "servers tuple types" to be consumed by the function.
- Use alias to give name to the fields of this type.

I'm giving an example on how it could look all together:

```
using hostname_t = std::string;
using port_t = int;
using read_only_t = int;

/**
 * @brief Type holding the required server info for the read_only Monitoring actions.
 */
using read_only_server_t = std::tuple<hostname_t,port_t,read_only_t>;

enum READ_ONLY_SERVER_T {
	HOSTNAME = 0,
	PORT,
	READONLY,
	__SIZE
};
```

This can be placed in the header. What do you think?

Reply:
Great!